### PR TITLE
feat: [GTM-1010]: add extra CCM telemetry events

### PR DESCRIPTION
**Pull request by nanaxuHarness**
##### Summary:

this is to add more CCM telemetry events for better data quality
1) add all CCM page events
2) add more CCM sign up flow track events
3) add a connector every wizard step track events

##### Jira Links:

https://harness.atlassian.net/browse/PLG-1010

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by nanaxuHarness** [comment]
retrigger jest

**Review by rishik30 — DISMISSED** [review]

File: src/modules/10-common/constants/TrackingConstants.ts [hunk 1]
```diff
@@ -47,7 +47,87 @@ export enum SecretActions {
 
 export enum ConnectorActions {
   StartCreateConnector = 'Start Create Connector',
```
Comment (nanaxuHarness) [inline src/modules/10-common/constants/TrackingConstants.ts:49]: Create Connector Started

File: src/modules/10-common/pages/pageContext/PageName.ts [hunk 2]
```diff
@@ -86,5 +86,25 @@ export enum PAGE_NAME {
   OnboardingPage = 'OnboardingPage',
   OnboardingDetailPage = 'OnboardingDetailPage',
   CFWorkflowsPage = 'CFWorkflowsPage',
-  SecretDetailsHomePage = 'SecretDetailsHomePage'
+  SecretDetailsHomePage = 'SecretDetailsHomePage',
+  CETrialHomePage = 'CETrialHomePage',
```
Comment (nanaxuHarness) [inline src/modules/10-common/pages/pageContext/PageName.ts:90]: CCM Trial Home

File: src/modules/10-common/constants/TrackingConstants.ts [hunk 3]
```diff
@@ -47,7 +47,87 @@ export enum SecretActions {
...
+  SetupEngineSubmit = 'Setup Engine Submitted',
+  AzureKeyValueFormLoad = 'Azure Key Value Form Loaded',
+  AzureKeyValueFormSubmit = 'Azure Key Value Form Submitted',
+  SetupVaultLoad = 'Setup Vault Loaded',
+  SetupVaultSubmit = 'Setup Vault Submitted',
+  OverviewLoad = 'Connector Overview Loaded',
+  OverviewSubmit = 'Connector Overview Submitted',
+  AzureConnectorBillingLoad = 'Azure Connector Billing Loaded',
+  AzureConnectorBillingSubmit = 'Azure Connector Billing Submitted',
+  CEGcpConnectorBillingExportLoad = 'CE Gcp Connector Billing Export Loaded',
```
Comment (nanaxuHarness) [inline src/modules/10-common/constants/TrackingConstants.ts:75]: CE => CCM

File: src/modules/10-common/constants/TrackingConstants.ts [hunk 4]
```diff
@@ -47,7 +47,87 @@ export enum SecretActions {
...
+  AppDynamics = 'AppDynamics',
+  Splunk = 'Splunk',
+  NewRelic = 'NewRelic',
+  Prometheus = 'Prometheus',
+  Dynatrace = 'Dynatrace',
+  SumoLogic = 'SumoLogic',
+  PagerDuty = 'PagerDuty',
+  ErrorTracking = 'ErrorTracking',
+  DataDog = 'Datadog',
+  CEK8 = 'CEK8sCluster',
```
Comment (nanaxuHarness) [inline src/modules/10-common/constants/TrackingConstants.ts:127]: keep CE here

File: src/modules/10-common/constants/TrackingConstants.ts [hunk 5]
```diff
@@ -47,7 +47,87 @@ export enum SecretActions {
...
+  SecretCreationStepSubmit = 'Secret Creation Step Submitted',
+  CENGAwsConnectorCostUsageReportLoad = 'CCM NG AWS Connector Cost Usage Report Loaded',
+  CENGAwsConnectorCostUsageReportSubmit = 'CCM NG AWS Connector Cost Usage Report Submitted',
+  CENGAwsConnectorCrossAccountRoleStep1Load = 'CCM NG AWS Connector Cross Account Role Step1 Loaded',
+  CENGAwsConnectorCrossAccountRoleStep1Submit = 'CCM NG AWS Connector Cross Account Role Step1 Submitted',
+  CENGAwsConnectorCrossAccountRoleStep2Load = 'CCM NG AWS Connector Cross Account Role Step2 Loaded',
+  CENGAwsConnectorCrossAccountRoleStep2Submit = 'CCM NG AWS Connector Cross Account Role Step2 Submitted'
+}
+
+export enum ConnectorTypes {
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/constants/TrackingConstants.ts:99]: we already have a connector types enum. why do we need to create a duplicate?
https://github.com/harness/harness-core-ui/blob/develop/src/modules/35-connectors/constants.ts#L64

File: src/modules/35-connectors/components/CreateConnector/AWSCodeCommitConnector/AWSCCAuthStep.tsx [hunk 6]
```diff
@@ -115,6 +117,15 @@ export default function AWSCCAuthStep(props: AWSCCAuthStepProps) {
     }
   }
 
+  const { trackEvent } = useTelemetry()
+  React.useEffect(() => {
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/components/CreateConnector/AWSCodeCommitConnector/AWSCCAuthStep.tsx:121]: this useEffect is repeated in a lot of places.
can we create a custom hook version of trackEvent, which includes the useEffect? Something like:

```
import { useTrackEvent } from '@common/hooks/useTelemetry`

useTrackEvent({ ... })
```

File: src/modules/35-connectors/components/CreateConnector/CEK8sConnector/SecretCreationStep.tsx [hunk 7]
```diff
@@ -53,9 +55,20 @@ const SecretCreationStep: React.FC<StepProps<StepSecretManagerProps> & SecretCre
   }
 
   const handleSubmit = () => {
+    trackEvent(ConnectorActions.SecretCreationStepSubmit, {
```
Comment (abhinavrastogi-harness) [inline src/modules/35-connectors/components/CreateConnector/CEK8sConnector/SecretCreationStep.tsx:58]: i hope `trackEvent` can never throw an exception... even if the n/w call fails or returns 400/404/500.
It'll block the actual user action.

File: src/modules/75-ce/modals/CETrialModal/useCETrialModal.tsx [hunk 8]
```diff
@@ -84,7 +94,16 @@ const CETrial: React.FC<CETrialModalData> = props => {
             <Text className={css.description}>{getString('ce.ceTrialHomePage.modal.description')}</Text>
           </Layout.Vertical>
           <Layout.Horizontal spacing="small" padding={{ top: 'large' }}>
-            <Button intent="primary" text={getString('continue')} onClick={onContinue} />
+            <Button
+              intent="primary"
+              text={getString('continue')}
+              onClick={() => {
+                trackEvent(CCMActions.CCMStartPlanContinue, {
```
Comment (abhinavrastogi-harness) [inline src/modules/75-ce/modals/CETrialModal/useCETrialModal.tsx:101]: trackEvent should be called in the `onContinue` function.. user can click multiple times on a button. I'm expecting that onContinue will handle that internally.

Comment (nanaxuHarness) [reply src/modules/75-ce/modals/CETrialModal/useCETrialModal.tsx:101]: the `onContinue` is also called whenever the dialog is closed, we only want to track it when user clicks on the button.
```
      <Dialog
        isOpen={true}
        enforceFocus={false}
        canOutsideClickClose={false}
        canEscapeKeyClose={false}
        onClose={onContinue}
        className={cx(css.dialog, css.ceTrial)}
      >
```

File: src/modules/35-connectors/components/CreateConnector/CEK8sConnector/SecretCreationStep.tsx (hunk 7 above)
Comment (nanaxuHarness) [reply src/modules/35-connectors/components/CreateConnector/CEK8sConnector/SecretCreationStep.tsx:58]: this is a very fair concern, I couldn't find very details of this from the website, but they do have this about retries: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/ajs-classic/#retries

File: src/modules/10-common/constants/TrackingConstants.ts (hunk 5 above)
Comment (nanaxuHarness) [reply src/modules/10-common/constants/TrackingConstants.ts:99]: still these 3 no mappings yet, so save them here.

File: src/modules/35-connectors/components/CreateConnector/AWSCodeCommitConnector/AWSCCAuthStep.tsx (hunk 6 above)
Comment (nanaxuHarness) [reply src/modules/35-connectors/components/CreateConnector/AWSCodeCommitConnector/AWSCCAuthStep.tsx]: updated

**Review by abhinavrastogi-harness — DISMISSED** [review]

File: src/modules/75-ce/RouteDestinations.tsx [hunk 9]
```diff
@@ -244,6 +249,7 @@ const CERoutes: React.FC = () => {
           sidebarProps={CESideNavProps}
           path={routes.toCECOCreateGateway({ ...accountPathProps, ...projectPathProps })}
           exact
+          pageName={PAGE_NAME.CECODashboardPage}
```
Comment (rishik30) [inline src/modules/75-ce/RouteDestinations.tsx:252]: this is a different page route. It's not a dashboard but the start of creating AS rules. We should have a different name for this page.

**Review by rishik30 — CHANGES_REQUESTED** [review]

File: src/modules/75-ce/RouteDestinations.tsx (hunk 9 above)
Comment (nanaxuHarness) [reply src/modules/75-ce/RouteDestinations.tsx]: good catch!

**Review by rishik30 — APPROVED** [review]

**Review by reetika04 — APPROVED** [review]

**Pull request merged**